src/constraints/field.rs

```diff
@@ -190,44 +190,32 @@ fn equal_cells<B: Backend>(
                 ConstOrCell::Cell(cvar) => *cvar,
             };
 
-            // compute the result
-            let res = compiler.backend.new_internal_var(
-                Value::Hint(Arc::new(move |backend, env| {
-                    let x1 = backend.compute_var(env, x1)?;
-                    let x2 = backend.compute_var(env, x2)?;
-                    if x1 == x2 {
-                        Ok(B::Field::one())
-                    } else {
-                        Ok(B::Field::zero())
-                    }
-                })),
-                span,
-            );
-
             // 1. diff = x2 - x1
             let neg_x1 = compiler.backend.neg(&x1, span);
             let diff = compiler.backend.add(&x2, &neg_x1, span);
```

Comment:
I think we could have

```rust
trait Backend {
  fn sub(a: Self::CellVar, b: Self::CellVar, span) -> Self::CellVar { Self::add(a, Self::neg(b, span), span) }
}
```

Reply:
That will be nicer.
I will implement https://github.com/zksecurity/noname/issues/56 in the [optimization PR](https://github.com/zksecurity/noname/pull/64) which can also wrap the constant as CellVar so that this new `sub` function can work with constant as well.